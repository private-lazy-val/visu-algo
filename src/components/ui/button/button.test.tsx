import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import { Button } from './button';
import renderer from 'react-test-renderer';

describe('Button component', () => {
    it('renders with text', () => {
        const tree = renderer
            .create(<Button text="Click me"/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders without text', () => {
        const tree = renderer
            .create(<Button />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders disabled button', () => {
        const tree = renderer
            .create(<Button disabled />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders with loading indicator', () => {
        const tree = renderer
            .create(<Button isLoader={true} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('calls onClick callback when clicked', () => {
        const callBack = jest.fn();
        render(<Button text="Click me" onClick={callBack}/>);

        fireEvent.click(screen.getByText('Click me'));
        expect(callBack).toHaveBeenCalledTimes(1);
    });
})