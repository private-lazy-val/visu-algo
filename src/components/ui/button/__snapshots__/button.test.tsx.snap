// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders disabled button 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Button component renders with loading indicator 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Button component renders with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Click me
  </p>
</button>
`;

exports[`Button component renders without text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
